### ISSUE TITLE
Wechsel von Mysql zu Postgresql

### ISSUE THREAD

**Issue by tabacha** [bensteffen/adfc-t30-api#6]
Für die Ermittlung des zuständigen Polizeireviers wäre es sinnvoll eine postgresql anzubinden.

**Comment by bensteffen** [comment]
Ok, ich schlage vor wir machen database/SqlConnection zu einer abstrakten Klasse mit abstrakter Methode "establish($credentials)". Und leiten davon eine Klasse für MySQL und PostgreSQL ab, die "estabish" entsprechend implementieren. Ich verwende $this->dbConnection->query($sqlQuery), um SQL auf der Verbindung auszuführen. Ich weiß nicht, ob das bei PostgreSQL genauso funktioniert?

**Comment by bensteffen** [comment]
Deswegen wäre es vlt. geschickt auch eine abstrakte Methode "executeQuery($sqlQuery)" hinzuzufügen. Dann kümmert sich jede Connection selbst darum, wie sie Queries ausführt.

**Comment by tabacha** [comment]
Ja, ungefähr so sollte man das machen. Wobei ich query eigentlich lieber vermeiden, weil es anfällig von SQL Injektion ist.

Eine alternative statt seine eigene DB-Abstaktionsschicht zu nutzen, wäre: PDO

https://www.php.net/manual/de/book.pdo.php

Tickets #1 #2 und #3 sind für mich aber von der Prio her wesentlich wichtiger. Das kann ich aber hier nicht taggen :-)
